pkg/cmd/codespace/ssh.go

```diff
@@ -298,6 +310,103 @@ func checkAndUpdateOldKeyPair(sshContext ssh.Context) *ssh.KeyPair {
 	return nil
 }
 
+// hasUploadedValidCustomKey checks which private keys are used in ssh configration and
+// compares them to uploaded public keys to see if there is a match
+func hasUploadedPublicKeyForConfig(
+	ctx context.Context,
+	sshContext ssh.Context,
+	apiClient apiClient,
+) bool {
+	configuredPrivateKeyPaths, err := getConfiguredPrivateKeys(ctx)
+	if err != nil {
+		return false
```

Comment:
A lot of the error handling is very basic/non-existent currently, once we're good on using this approach I'll circle back and polish it